src/parsers/manifestjson.js

```diff
@@ -342,8 +362,14 @@ export default class ManifestJSONParser extends JSONParser {
     ) {
       baseObject = messages.MANIFEST_BAD_HOST_PERMISSION;
       overrides.message = `Permissions ${error.message}.`;
-    } else if (error.keyword === 'type') {
+    } else if (error.keyword === SCHEMA_KEYWORDS.TYPE) {
       baseObject = messages.MANIFEST_FIELD_INVALID;
+    } else if (error.keyword === SCHEMA_KEYWORDS.PRIVILEGED) {
+      baseObject = error.params?.isPrivilegedAddon
```

Comment:
In a previous version (I think one that only existed locally on my machine) the error reported on non privileged addons did not have any params at all, but then I changed it to always include the params (mainly because it seems more consistent to make sure that the error collected by a certain custom keyword has always the same format).

And so in practice error.params should always be defined with a keyword set to SCHEMA_KEYWORDS.PRIVILEGED, and so I'll just be removing the unnecessary optional chaining from this one.

I think that for L336 I may have not being passing through such an intermediate version (both the errors did have the exact same privilegedPermissions property set in the error params, and that is likely the only reason why the optional chaining was left only in this one.